src/main/index.js

```diff
@@ -165,13 +157,29 @@ async function startup ({
     createMenu(ipc)
     backgroundProcesses.addClient('mapeoCore', port2)
 
+    await Promise.all([
+      // Create folders for data, custom presets, and custom styles
+      mkdirp(datadir),
+      mkdirp(mapsdir),
+      mkdirp(path.join(userDataPath, 'presets')),
+      mkdirp(path.join(userDataPath, 'styles'))
+    ])
+
+    // Running this in the main thread rather than in a background process because
+    // map-server uses Node Worker Threads, and the background processes actually
+    // run in a browser window (with Node integration turned on) but they do not
+    // support Node workers (only Web Workers). The reason for running the other
+    // processes (map printer and mapeo core) in a background process was because
+    // processes in the main thread are blocking for the render thread.
+    // Fortunately map server does not have any expensive functions so it should
+    // not slow down the main process nor block the render thread if we run it
+    // here from the main process...
+    mapServer = createMapServer(undefined, {
+      database: new Database(path.join(mapsdir, 'maps.db'))
+    })
```

Comment:
maybe out of scope, but wondering if this should point to `tmp-resources` when in development, instead of the user data directory

Reply:
yeah some kind of tmp directory would make sense. might be good to make an issue for it rather than deal with it in this pr.

Reply:
yeah we already create a `tmp-directory` which includes some assets like the presets. just a matter of whether it makes sense to put the maps db here during dev too